### PR TITLE
chore(node): add stub modules

### LINKED ISSUE #1734: node: Add stubs for APIs in several modules
We should add stubs for following modules, so we can set them up automatically in REPL when running in `compat` mode:

- [ ] cluster
- [ ] domain
- [ ] https
- [ ] tls
- [ ] v8

The idea is to add top level export that would throw `not implemented` error instead of throwing the error when the module is loaded.

Ref https://github.com/denoland/deno/pull/13127

### PR DESCRIPTION
**Pull request by kt3k**
This PR adds stub implementations of missing node modules.

closes #1734

### CODE REVIEW

File: node/http2.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+}
+export class Http2SecureServer {
+  constructor() {
+    notImplemented();
+  }
+}
+export function createServer() {}
+export function createSecureServer() {}
+export function connect() {}
+export const constants = undefined;
```
Comment (bnoordhuis) [inline node/http2.ts:49]:
```suggestion
export const constants = {};
```
(just a suggestion, `undefined` isn't really wrong either)

File: node/http2.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+export function getDefaultSettings() {}
+export function getPackedSettings() {}
+export function getUnpackedSettings() {}
+export const sensitiveHeaders = undefined;
```
Comment (bnoordhuis) [inline node/http2.ts:53]:
```suggestion
export const sensitiveHeaders = Symbol("nodejs.http2.sensitiveHeaders");
```

File: node/repl.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+// Copyright Joyent and Node contributors. All rights reserved. MIT license.
+
+import { notImplemented } from "./_utils.ts";
+
+class REPLServer {
+  constructor() {
+    notImplemented();
+  }
+}
+export const builtinModules = undefined;
```
Comment (bnoordhuis) [inline node/repl.ts:11]:
```suggestion
export const builtinModules = ["assert", "async_hooks", "buffer", "child_process", "cluster", "console", "constants", "crypto", "dgram", "diagnostics_channel", "dns", "domain", "events", "fs", "http", "http2", "https", "inspector", "module", "net", "os", "path", "perf_hooks", "process", "punycode", "querystring", "readline", "repl", "stream", "string_decoder", "sys", "timers", "tls", "trace_events", "tty", "url", "util", "v8", "vm", "wasi", "worker_threads", "zlib"];
```
Which, granted, is a bit of a lie.

File: node/tls.ts [hunk 4]
```diff
@@ -1 +1,47 @@
...
+export function createSecurePair() {
+  notImplemented();
+}
+export function createServer() {
+  notImplemented();
+}
+export function getCiphers() {
+  notImplemented();
+}
+export const rootCertificates = undefined;
```
Comment (bnoordhuis) [inline node/tls.ts:28]: Side discussion: open question how to support this. Node uses a hard-coded list of root certificates, based off Mozilla's NSS certdata.txt.

We use `rustls_native_certs::load_native_certs()` but that produces different results on different platforms. For reasons of compatibility, we may want to use node's list instead.

**Review by bnoordhuis — APPROVED** [review]
👍

File: node/repl.ts (hunk 3 above)
Comment (kt3k) [reply node/repl.ts:11]: Looks good to me. Thanks for the suggestion!

File: node/tls.ts (hunk 4 above)
Comment (kt3k) [reply node/tls.ts:28]: maybe we should add an option to `Deno.ConnectTlsOptions` for completely swap the root certificates? https://doc.deno.land/deno/stable/~/Deno.ConnectTlsOptions

Comment (bnoordhuis) [reply node/tls.ts:28]: Yes, I agree. That'd be useful for more than just node compat too.

Comment (lucacasonato) [reply node/tls.ts:28]: @bnoordhuis `rustls_native_certs::load_native_certs()` no we don't. We use the same mozilla list by default. You can switch to your native cert list by setting `DENO_TLS_CA_STORE=system` or `DENO_TLS_CA_STORE=system,mozilla` to use both.

Comment (bnoordhuis) [reply node/tls.ts:28]: Oh, that's right - I believe I even reviewed the PR that added that. 🤦

Then perhaps we want to bypass `DENO_TLS_CA_STORE=system` and stick to the built-in list.

**Review by bartlomieju — APPROVED** [review]
LGTM!

**Pull request merged**